pkg/reconciler/domainmapping/controller.go

```diff
@@ -72,7 +75,15 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 	certificateInformer.Informer().AddEventHandler(handleControllerOf)
 	ingressInformer.Informer().AddEventHandler(handleControllerOf)
 
-	r.resolver = resolver.NewURIResolver(ctx, impl.EnqueueKey)
+	dmTracker := tracker.New(impl.EnqueueKey, controller.GetTrackerLease(ctx))
+	dmInformerFactory := &pkgapisduck.CachedInformerFactory{
+		Delegate: &pkgapisduck.EnqueueInformerFactory{
+			Delegate:     addressable.Get(ctx),
+			EventHandler: controller.HandleAll(dmTracker.OnChanged),
+		},
+	}
+
+	r.resolver = resolver.NewURIResolverWithControllerContext(ctx, dmInformerFactory, dmTracker)
```

Comment:
I might be missing something, but it **looks** like, this could just stay `NewURIResolver` as the setup looks identical. You seem to have adjusted `NewURIResolver` to capture the controller context and ignore the `ReconcileKind` context for lister fetching, so we should be golden without this new function, I think?

Reply:
@markusthoemmes That's true, sorry about that!